books/free-programming-books-langs.md

```diff
@@ -790,27 +790,32 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [A git Primer](https://danielmiessler.com/study/git/) - Daniel Miessler
 * [A Visual Git Reference](http://marklodato.github.io/visual-git-guide/index-en.html) - Mark Lodato
 * [Conversational Git](http://blog.anvard.org/conversational-git/)
+* [Distributed Version Control with Git: Mastering the Git command line - Third Edition](https://www.pdfdrive.com/distributed-version-control-with-git-mastering-the-git-command-line-third-edition-e157843022.html) - Lars Vigel & Alex Blewitt
```

Comment:
In general, we do not link to file sharing sites such as pdfdrive, or file hosting sites such as google docs. or dropbox.